wp_api/src/templates.rs

```diff
@@ -0,0 +1,225 @@
+use crate::{
+    impl_as_query_value_from_as_str,
+    post_types::PostType,
+    posts::PostId,
+    url_query::{
+        AppendUrlQueryPairs, AsQueryValue, FromUrlQueryPairs, QueryPairs, QueryPairsExtension,
+        UrlQueryPairsMap,
+    },
+    EnumFromStrParsingError, UserId,
+};
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+use strum_macros::IntoStaticStr;
+use wp_contextual::WpContextual;
+
+#[derive(
+    Debug,
+    Default,
+    Clone,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    Serialize,
+    Deserialize,
+    uniffi::Enum,
+)]
+#[serde(rename_all = "snake_case")]
+pub enum TemplateStatus {
+    Draft,
+    Future,
+    Pending,
+    Private,
+    #[default]
+    Publish,
+    #[serde(untagged)]
+    Custom(String),
+}
+
+impl_as_query_value_from_as_str!(TemplateStatus);
+
+impl TemplateStatus {
+    pub fn as_str(&self) -> &str {
+        match self {
+            Self::Draft => "draft",
+            Self::Future => "future",
+            Self::Pending => "pending",
+            Self::Private => "private",
+            Self::Publish => "publish",
+            Self::Custom(status) => status,
+        }
+    }
+}
+
+impl FromStr for TemplateStatus {
+    type Err = EnumFromStrParsingError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "draft" => Ok(Self::Draft),
+            "future" => Ok(Self::Future),
+            "pending" => Ok(Self::Pending),
+            "private" => Ok(Self::Private),
+            "publish" => Ok(Self::Publish),
+            value => Ok(Self::Custom(value.to_string())),
+        }
+    }
+}
```

Comment:
Use `strnum` for consistence

Reply:
Right. I implemented this before the `strum` changes and didn't think of it when I picked up the branch again. Addressed in 8d5958e0304cd2a1c56e9459cb6884d410787db4.